src/components/input/_macro.njk

```diff
@@ -110,18 +110,23 @@
         {% endif %}
     {% endset %}
 
-    {% if params.charCheck is defined and params.charCheck.limit %}
-        {% call onsCharLimit({
-            "id": params.id,
-            "limit": params.charCheck.limit,
-            "charCountSingular": params.charCheck.charCountSingular,
-            "charCountPlural": params.charCheck.charCountPlural,
-            "charCountOverLimitSingular": params.charCheck.charCountOverLimitSingular,
-            "charCountOverLimitPlural": params.charCheck.charCountOverLimitPlural
-        }) %}
-            {{ field | safe }}
-        {% endcall %}
-    {% elif params.mutuallyExclusive %}
+    {% if params.charCheck is defined %}
+        {% set charCheckField %}
+            {% call onsCharLimit({
+                "id": params.id ~ "-check",
+                "limit": params.charCheck.limit,
+                "type": "limit",
```

Comment:
Think you can get rid of this?

Reply:
That was put in because Input was using the charCheck macro but textarea was doing something similar but not using the macro. To standerdise the way charCheck was implemented I managed to get them both using the charCheck macro. But there were parts of the macro that wasn't needed for the textarea implementation.
So I used that variable to differentiate between using it on an input or textarea. Do you think we should go back to the way textareas were working originally? Or I could rename the variable? It was just to standerdise the way charChecks were being used

Reply:
Right ok that sounds good. Does the textarea use it's own JS still or does it use the charcheck?

Reply:
Nah it still uses its own JS, maybe I should move the charcheck stuff out of input into its own folder and intergrate it with the textarea JS

Reply:
On textarea/mutually exclusive components limits are defined using the maxlength, charCountSingular and charCountPlural keys. 

Is there a reason why input can't use the same keys without the charCheck object? Otherwise we have to work round this difference in runner's templates which is why I think we may have this bug at the moment.

Reply:
Because input and textarea are two different components I think getting both to use the charCheck macro rather than having them each use their own implementation like textarea currently does makes sense to me as it will standardise the way character limits are handled.
The reason why we have the bug at the moment is because adding a char limit and mutually exclusive to inputs is currently mutually exclusive in itself. So you can either add one or the other. The changes I've made in this PR allow you to use both.

Reply:
To make it a bit easier for you to implement in runner. Ive got Input and Textarea both using `limit` now instead of one using `maxlength` and one using `limit`. I tried looking into getting the other ones the same too but there needs to be a charCountSingular and a charCountOverLimitSingular param because of the need for messages for over and under the char count as in [this](http://localhost:3030/components/search/examples/search-character-check) search example. I feel like this is the best solution to fix the bug and maintain as much consistency as possible.